File: basic/src/main/java/top/non/late/pavilion/basic/java/designs/builder/code/JVMBookBuilder.java
package top.non.late.pavilion.basic.java.designs.builder.code;

/**
 * Created by nonlatepavilion@outlook.com on 2017/9/7.
 * Desc: jvm book Builder
 */
public class JVMBookBuilder implements BookBuilder {
    private BookProduct product;
    public JVMBookBuilder() {
        this.product = new BookProduct();
    }
    @Override
    public void bookName() {
        this.product.setBookName("深入理解Java虚拟机");
    }

    @Override
    public void author() {
        this.product.setAuthor("周志明");
    }

    @Override
    public void isbn() {
        this.product.setIsbn("9787111421900");
    }

    @Override
    public BookProduct build() {
        return product;
    }
}

File: basic/src/main/java/top/non/late/pavilion/basic/kotlin/designs/builder/JVMBookBuilder.kt
package top.non.late.pavilion.basic.kotlin.designs.builder

/**
 * Created by nonlatepavilion@outlook.com on 2017/9/25.
 * Desc: 
 */
class JVMBookBuilder : BookBuilder {
    private val product = BookProduct()
    override fun bookName() {
        product.bookName = "深入理解Java虚拟机"
    }

    override fun author() {
        product.author = "周志明"
    }

    override fun isbn() {
        product.isbn = "9787111421900"
    }

    override fun build(): BookProduct {
        return product
    }
}